File: Leetcode/PrefixSum/1109.py
class Solution:
    def corpFlightBookings(self, bookings: List[List[int]], n: int) -> List[int]:
        counter = [0] * (n + 1)

        for i, j, k in bookings:
            counter[i - 1] += k
            if j < n: counter[j] -= k
        for i in range(n + 1):
            counter[i] += counter[i - 1]
        return counter[:-1]

File: Leetcode/PrefixSum/1109.js
/**
 * @param {number[][]} bookings
 * @param {number} n
 * @return {number[]}
 */
// 暴力模拟法，可AC
var corpFlightBookings = function (bookings, n) {
  const arr = new Array(n).fill(0);
  for (let [i, j, k] of bookings) {
    while (i <= j) {
      arr[i - 1] += k;
      i++;
    }
  }
  return arr;
};

// 前缀和方法
/**
 * @param {number[][]} bookings
 * @param {number} n
 * @return {number[]}
 */
var corpFlightBookings = function (bookings, n) {
  const arr = new Array(n).fill(0);
  for (let [i, j, k] of bookings) {
    arr[i - 1] += k;
    if (j < n) arr[j] -= k;
  }
  for (let i = 1; i < n; i++) {
    arr[i] += arr[i - 1]
  }
  return arr;
};